ext/RMagick/rmimage.c

```diff
@@ -3609,16 +3923,31 @@ composite_tiled(int bang, int argc, VALUE *argv, VALUE self)
     status = MagickTrue;
     columns = comp_image->columns;
 
+#if defined(IMAGEMAGICK_7)
+    exception = AcquireExceptionInfo();
+#endif
+
     // Tile
     for (y = 0; y < (long) image->rows; y += comp_image->rows)
     {
         for (x = 0; status == MagickTrue && x < (long) image->columns; x += columns)
         {
+#if defined(IMAGEMAGICK_7)
+            BEGIN_CHANNEL_MASK(image, channels);
+            status = CompositeImage(image, comp_image, MagickTrue, operator, x, y, exception);
+            END_CHANNEL_MASK(image);
+            rm_check_exception(exception, image, bang ? RetainOnError: DestroyOnError);
+#else
             status = CompositeImageChannel(image, channels, operator, comp_image, x, y);
             rm_check_image_exception(image, bang ? RetainOnError: DestroyOnError);
```

Comment:
if exception was raised in `rm_check_image_exception()`, it can not release a heap area of `exception`.

Reply:
But `exception` was created inside an `#if defined(IMAGEMAGICK_7)` block?